lib/server/push.api.js

```diff
@@ -443,10 +443,16 @@ Push.Configure = function(options) {
 
     Meteor.setInterval(function() {
 
-        if (!isSendingNotification) {
-            // Set send fence
-            isSendingNotification = true;
+        if (isSendingNotification) {
+            return;
+        }
+        // Set send fence
+        isSendingNotification = true;
 
+        var countSent = 0;
+        var batchSize = options.sendBatchSize || 1;
+        while (Push.notifications.find().count() > 0 && countSent++ < batchSize) {
```

Comment:
we can't rely on notifications being removed - theres an option for keeping sent notes. We could how ever do something like:
```js
// Find notifications that is not being or already sent
var pendingNotifications = Push.notifications.find({ $and: [
    // Message is not sent
    { sent : { $ne: true } },
    // And not being sent by other instances
    { sending: {$ne: true} },
  ]}, {
    // Sort by created date
    sort: { createdAt: 1 },
    limit: batchSize
  });

pendingNotifications.forEach(function(notification) {
  /* reserve and send... */
});
```

--

Mental note: We should ensure index on `createdAt` etc.